optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/common/decorator/WeightFactorySelectionSorter.java

```diff
@@ -34,13 +34,15 @@
  * Sorts a selection {@link List} based on a {@link SelectionSorterWeightFactory}.
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  * @param <T> the selection type
+ * @param <W> the weight type
  */
-public class WeightFactorySelectionSorter<Solution_, T> implements SelectionSorter<Solution_, T> {
+public class WeightFactorySelectionSorter<Solution_, T, W extends Comparable<W>>
+        implements SelectionSorter<Solution_, T> {
 
-    private final SelectionSorterWeightFactory<Solution_, T> selectionSorterWeightFactory;
-    private final Comparator<Comparable> appliedWeightComparator;
+    private final SelectionSorterWeightFactory<Solution_, T, W> selectionSorterWeightFactory;
+    private final Comparator<W> appliedWeightComparator;
```

Comment:
But I don't want to push you to merging this :-) Unless there is a consensus, it's OK to close this or continue discussion.